File: cse465-ComparativeProgramming/Homework 5 (Chapts 5&6, Reflection, triangular matrix access function)/rand_bench/rand.c
#include <iostream>
#include <cstdlib>
using namespace std;

int main() {
	double* doubles = new double[1000];
	for( int i = 0; i < 1000; i++) {
		doubles[i] = -1 + ((double)rand()) * 2;
		//cout << doubles[i] << "\n";
	}

	double sum = 0;

	for(int i = 0; i < 1000; i++)
		sum += doubles[i];

	cout << sum << "\n";
	delete doubles;
}

File: cse465-ComparativeProgramming/Homework 5 (Chapts 5&6, Reflection, triangular matrix access function)/rand_bench/rand.cs
using System;

namespace Ran
{
    class Program
    {
        static void Main(string[] args)
        {
            double[] doubles = new double[1000000];
            Random random = new Random();
            for (int i = 0; i < 1000000; i++)
                doubles[i] = random.NextDouble();
            double sum = 0;
            for (int i = 0; i < 1000000; i++)
                sum += doubles[i];
            Console.WriteLine(sum);
        }
    }
}

